src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1755,5 +1755,117 @@ public static void ZeroFreeGlobalAllocUnicode(IntPtr s)
             RuntimeImports.RhZeroMemory(s, (UIntPtr)(Win32Native.lstrlenW(s) * 2));
             FreeHGlobal(s);
         }
+
+        /// APIs for managing Native Libraries 
+
+        /// <summary>
+        /// NativeLibrary Loader: Simple API
+        /// This method is a wrapper around OS loader, using "default" flags.
+        /// </summary>
+        /// <param name="libraryName">The name of the native library to be loaded</param>
+        /// <returns>The NativeLibrary object for the loaded library</returns>  
+        /// <exception cref="System.DllNotFoundException ">Thrown if the library can't be found.</exception>
+        /// <exception cref="System.BadImageFormatException">Thrown if the library is not valid.</exception>
+        public static IntPtr LoadLibrary(string libraryPath)
+        {
+            return LoadLibraryFromPath(libraryPath, true);
+        }
+
+        /// <summary>
+        /// NativeLibrary Loader: Simple API that doesn't throw
+        /// </summary>
+        /// <param name="libraryName">The name of the native library to be loaded</param>
+        /// <param name="handle">The out-parameter for the loaded NativeLibrary object</param>
+        /// <returns>True on successful load, false otherwise</returns>  
+        public static bool TryLoadLibrary(string libraryPath, out IntPtr handle)
+        {
+            handle = LoadLibraryFromPath(libraryPath, false);
+            return handle != IntPtr.Zero;
+        }
+
+        /// <summary>
+        /// NativeLibrary Loader: High-level API
+        /// Given a library name, this function searches specific paths based on the 
+        /// runtime configuration and attributes of the calling module.
+        /// This LoadLibrary() method does not invoke the managed call-backs for native library resolution: 
+        /// * AssemblyLoadContext.LoadUnmanagedDll()
+        /// </summary>
+        /// <param name="libraryName">The name of the native library to be loaded</param>
+        /// <param name="dllImportSearchPath">The search path</param>
+        /// <param name="assembly">The assembly loading the native library</param>
+        /// <returns>The NativeLibrary object for the loaded library</returns>  
+        /// <exception cref="System.DllNotFoundException ">Thrown if the library can't be found.</exception>
+        /// <exception cref="System.BadImageFormatException">Thrown if the library is not valid.</exception>        
+        public static IntPtr LoadLibrary(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
```

Comment:
This API is weird.  Why make an enum value null? Can we just use the default value expected? In this case `DllImportSearchPath.LegacyBehavior`?

Reply:
Wasn't the `DllImportSearchPath?` meant to distinguish between the case where you actually want the legacy behavior and the case where you want the flags to be fetched from the assembly?

As the code is written, we do not really need the Nullable there. Folks can just pass  `default(DllImportSearchPath)` to get the default.

cc @GrabYourPitchforks

Reply:
Legacy shouldn't be the default since it's insecure. Null meaning check assembly attributes could be ok, but if the assembly doesn't have attributes, I'm not sure what we do. What's the problem with just requiring the user to choose?

Reply:
Sure, I'll change the semantics of DllImportSearchPath being null to mean that the DllImportSearchPathFlags should be read from the module (and not LegacyBehavior).

This fits well with the existing code, because current implementation is
- Look up the `DefaultDllImportSearchPathAttribute` on the `PInvoke` is available use-it. With the `Marshall.LoadLibrary()` API this will mean if `DllImportSearchPathFlag` parameter is specified use it.
- If not, Look up the `DefaultDllImportSearchPathAttribute` on the module, and use it if that's available.
- If not, use default behavior specified [here](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.defaultdllimportsearchpathsattribute?redirectedfrom=MSDN&view=netframework-4.7.2)